android/src/XCSoar.java

```diff
@@ -243,8 +245,13 @@ private void requestAllPermissions() {
        permissions before using them; mentioning them in the manifest
        is not enough */
 
-    checkRequestPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE);
-    checkRequestPermission(Manifest.permission.ACCESS_FINE_LOCATION);
+    if (!hasAllPermissions()) {
+      try {
+        this.requestPermissions(NEEDED_PERMISSIONS, 0);
+      } catch (Exception e) {
+        Log.e(TAG, "could not request permissions: " + String.join(", ", NEEDED_PERMISSIONS));
```

Comment:
Bad log message - it doesn't show the error details, and instead shows a list of permissions, but is unable to point out which failed.

Reply:
> unable to point out which failed

As I understand it only the whole call can fail.
If a permission is denied no exception would be thrown, but `onRequestPermissionsResult` would be called with a corresponding grantResult entry.
An exception would only occur when requestCode is negative, which doesn't apply because it is fixed to `0`.

Reply:
The code which you removed logs the Exception object. Yours does not.

Reply:
Ah okay, I added the exception object back to the log message.